### PR TITLE
Add protocols for Plugins

### PR DESCRIPTION
**Pull request by ps2**

### CODE REVIEW

File: LoopKit/LoopPlugin.swift [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+//  LoopKit
+//
+//  Created by Pete Schwamb on 7/24/19.
+//  Copyright © 2019 LoopKit Authors. All rights reserved.
+//
+
+import Foundation
+
+public enum LoopPluginBundleKey: String {
+    case pumpManagerDisplayName = "com.loopkit.Loop.PumpManagerDisplayName"
```
Comment (darinkrauss) [inline LoopKit/LoopPlugin.swift:12]: Is this really `com.loopkit.Loop.*` or `com.loopKit.LoopKit.*`?

File: LoopKit/LoopPlugin.swift [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+//  LoopPlugin.swift
+//  LoopKit
+//
+//  Created by Pete Schwamb on 7/24/19.
+//  Copyright © 2019 LoopKit Authors. All rights reserved.
+//
+
+import Foundation
+
+public enum LoopPluginBundleKey: String {
```
Comment (darinkrauss) [inline LoopKit/LoopPlugin.swift:11]: Presumably we'll need to add a ServiceManager version of these at some point, yes?

**Review by darinkrauss — APPROVED** [review]
LGTM with only a couple minor questions. 🚀

File: LoopKit/LoopPlugin.swift (hunk 2 above)
Comment (ps2) [reply LoopKit/LoopPlugin.swift:11]: Yes, I believe so.

File: LoopKit/LoopPlugin.swift (hunk 1 above)
Comment (ps2) [reply LoopKit/LoopPlugin.swift:12]: I think it's actually Loop specific. The plugin protocols, along with CGMManager and PumpManager protocol (and associated UI version) could at some point be moved out into a separate set of frameworks (LoopInterfaces.framework and LoopUIInterfaces.framework, maybe?).  LoopKit is a toolkit for building insulin delivery apps, and defining the interfaces that the very specific Loop app supports is really a separate concern.

**Pull request merged**